### ISSUE TITLE
Merge kalibrate-hackrf with kalibrate-rtl?

### ISSUE THREAD

**Issue by rxseger** [scateu/kalibrate-hackrf#9]
Would it be possible to merge the two kalibrate's for hackrf and rtl into one project supporting both SDRs?

kalibrate-rtl: https://github.com/steve-m/kalibrate-rtl - last updated 2013, but includes several open pull requests: https://github.com/steve-m/kalibrate-rtl/pull/5 https://github.com/steve-m/kalibrate-rtl/pull/6 https://github.com/steve-m/kalibrate-rtl/pull/9

kalibrate-rtl is also included in Kali Linux: http://tools.kali.org/wireless-attacks/kalibrate-rtl - but not http://tools.kali.org/wireless-attacks/kalibrate-hackrf

I was about to try to merge kalibrate-rtl and kalibrate-hackrf, but even though they both descended from kalibrate 0.4.1 (originally from http://thre.at/kalibrate/kal-v0.4.1.tar.bz2 but now thre.at is down), the code formatting style is different (whitespace) for some reason making this non-trivial (but did merge the -rtl PR's into: https://github.com/rxseger/kalibrate-hackrf). Which should be considered the "original" fork to merge the other one into?

**Comment by rxseger** [comment]
Adding RTL-SDR support may also help in investigating the cause of https://github.com/scateu/kalibrate-hackrf/issues/6 Is it right (wildly inconsistent results on different channels, e.g. -33.570 ppm vs 38.369 ppm), for comparison.

to do: get a cheap rtl-sdr dongle to test with (first kalibrate-rtl, then work on merging into -hackrf)

**Comment by rxseger** [comment]
Instead of platform-specific SDR libraries, consider using a vendor-neutral library, e.g. [SoapySDR](https://github.com/pothosware/SoapySDR) (as used by [CubicSDR](https://github.com/cjcliffe/CubicSDR), even has [Homebrew recipes](https://github.com/pothosware/homebrew-pothos/wiki))

**Comment by Romeo-Golf** [comment]
I agree it would be useful to have a version of calibrate which combines any/all receivers currently supported on separate forks... And hopefully this would allow any future development work to benefit users of all of these platforms :)
